10up/ElasticPress
# Feature: Dynamic index

### Description of the Change
This PR adds a basic algorithm is to increase/decrease the number of posts/documents while indexing by timing each request. This also fixes the issue with very large documents (over 100MB) which exceed the `max_content_length` of Elasticsearch.



### Alternate Designs
n/a


### Benefits
A more reliable sync process that can self-tune base on the network and hardward condition at the sync time.



### Possible Drawbacks
None that I'm aware of.


### Verification Process
- Set bulk index setting to `1` and run sync. Check the setting value after the sync, see it changed.
- Upload some large documents then turn on document indexing. Run a sync, see the index succeeds.



### Checklist:






### Applicable Issues



### Changelog Entry
